cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/sqlite/SQLQueueCallable.java

```diff
@@ -22,54 +22,59 @@
  *
  * @api_private
  */
-public abstract class SQLQueueCallable<T> implements Callable<T> {
+public class SQLQueueCallable<T> implements Callable<T> {
 
     private SQLDatabase db;
-    private boolean runInTransaction = false;
+    private boolean runInTransaction;
+    private SQLCallable<T> sqlCallable;
+
+    /**
+     * <p>
+     *     Create a SQLQueueCallable object which will invoke the SQLCallable on the SQLDatabase
+     * </p>
+     * <p>
+     *     The SQLCallable will not be invoked in a new transaction
+     * </p>
+     * @param db The SQLDatabase to use for invoking
+     * @param sqlCallable The SQLCallable to invoke
+     */
+    SQLQueueCallable(SQLDatabase db, SQLCallable<T> sqlCallable) {
+        this.db = db;
+        this.sqlCallable = sqlCallable;
+    }
+
+    /**
+     * <p>
+     *     Create a SQLQueueCallable object which will invoke the SQLCallable on the SQLDatabase
+     * </p>
+     * <p>
+     *     The SQLCallable will be invoked in a new transaction if runInTransaction is true
+     * </p>
+     * @param db The SQLDatabase to use for invoking
+     * @param runInTransaction Whether to invoke the SQLCallable in a new transaction
+     * @param sqlCallable The SQLCallable to invoke
+     */
+    SQLQueueCallable(SQLDatabase db, boolean runInTransaction, SQLCallable<T> sqlCallable) {
```

Comment:
Small stylistic thing, shouldn't `runInTransaction` be the final argument so its easier to recognise thats its method overloading?